tartiflette_aiohttp/__init__.py

```diff
@@ -158,11 +158,6 @@ def register_graphql_handlers(
     if context_factory is None:
         context_factory = default_context_factory
 
-    if not iscoroutinefunction(context_factory):
-        raise Exception(
-            "`context_factory` parameter should be a coroutine function."
-        )
-
```

Comment:
I'm removing this check for two reasons:

1. I couldn't come up with a reliable way of checking whether context_factory is a coroutinefunction OR an asynccontextmanager, without first calling it.
2. It's not very Pythonic to do this sort of up-front type-checking. Better to let it just fail later on if the incorrect type.